File: articles/mental-accounting-personal-finance/fortran/mental_accounting_decision.f90
program mental_accounting_decision
  implicit none
  real :: debt_interest_rate, savings_rate, repayment_amount, label_penalty, gain
  integer :: decision

  debt_interest_rate = 0.22
  savings_rate = 0.02
  repayment_amount = 1000.0
  label_penalty = 150.0

  gain = (debt_interest_rate - savings_rate) * repayment_amount

  if (gain > label_penalty) then
    decision = 1
  else
    decision = 0
  end if

  print *, "Repay debt from labeled savings:", decision
end program mental_accounting_decision

File: articles/mental-accounting-personal-finance/c/mental_accounting_decision.c
#include <stdio.h>

double debt_repayment_gain(double debt_interest_rate, double savings_rate, double repayment_amount) {
    return (debt_interest_rate - savings_rate) * repayment_amount;
}

int repay_debt_from_labeled_savings(double debt_interest_rate, double savings_rate, double repayment_amount, double label_penalty) {
    double gain = debt_repayment_gain(debt_interest_rate, savings_rate, repayment_amount);
    return gain > label_penalty;
}

int main(void) {
    int decision = repay_debt_from_labeled_savings(0.22, 0.02, 1000.0, 150.0);
    printf("Repay debt from labeled savings: %d\n", decision);
    return 0;
}
